pepper-apis/src/main/java/org/broadinstitute/ddp/event/dsmtask/api/DsmTaskProcessorAbstract.java

```diff
@@ -0,0 +1,19 @@
+package org.broadinstitute.ddp.event.dsmtask.api;
+
+import static org.broadinstitute.ddp.event.dsmtask.api.DsmTaskResultData.DsmTaskResultType.ERROR;
+import static org.broadinstitute.ddp.event.dsmtask.api.DsmTaskResultData.DsmTaskResultType.SUCCESS;
+
+public abstract class DsmTaskProcessorAbstract implements DsmTaskProcessor {
+
+    @Override
+    public DsmTaskResultData processDsmTask(DsmTaskData dsmTaskData) {
+        try {
+            doIt(dsmTaskData);
+        } catch (Exception e) {
+            return new DsmTaskResultData(ERROR, e.getMessage(), dsmTaskData);
```

Comment:
Should an error here result in `ack()` or `nack()`?  Or is it up to the implementing class?

Reply:
Also, should we pass back the entire exception object instead of just the message? So someone can log the exception and it's stacktrace?

Reply:
The idea is that in any case the request considered as processed (ack) and to the result topic sent a message with results (SUCCESS or ERROR).
@yufengwng are you suggest to send a result message a stacktrace?

Reply:
What I mean was, if there was an exception while processing, we should log it and try again. So someone should log the exception and call `nack()` so pubsub will send the message again and we can try processing again (e.g. imagine if there was a temporary database connection issue).

However, if the message was faulty and we're unable to process it due to "client mistake" then yeah, I can see us returning some sort of "error" and then `ack()` the message. For example, imagine if we got an invalid email -- that will be improper data and a client error.

So I guess what we're missing in implementation is a way to distinguish between "client error" versus "server processing error". In the client case, we should probably `ack()` message and move on. In server case, we should probably `nack()` and try again.

Reply:
I added the nack() for the case when internal server error happened in auth0.
About temporary DB connection issue - I need to investigate more what kind of errors are possible..